### PR TITLE
AKS and EKS modules

### PR DESCRIPTION
**Pull request by cicharka**
Using AKS and EKS Terraform configuration directly with Epiphany.
Result of Spike [#2982](https://github.com/epiphany-platform/epiphany/issues/2982)

### CODE REVIEW

File: eks/modules/nodes/variables.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+variable "subnet_ids" {
+  description = "Subnet ids to join to"
+  type = list(string)
+}
+
+variable "disk_size" {
+  description = "Disk size"
+  type        = number
+}
```
Comment (erzetpe) [inline eks/modules/nodes/variables.tf:25]: Maybe we want to add some validations here?

File: eks/modules/nodes/variables.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+variable "ami_type" {
+  description = "Type of Amazon Machine Image (AMI) associated with the EKS Node Group"
+  type        = string
+}
+
+variable "ec2_ssh_key" {
+  description = "EC2 Key Pair name that provides access for SSH communication with the worker nodes in the EKS Node Group"
+  type        = string
+}
```
Comment (erzetpe) [inline eks/modules/nodes/variables.tf:35]: And here.

File: eks/eks_variables.tf [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+variable "k8s_version" {
+  description = "Kubernetes version to install"
+  type        = string
+  default     = "1.22"
+}
+
+variable "autoscaler_version" {
+  description = "Kubernetes autoscaler version"
+  type        = string
+  default     = null
```
Comment (erzetpe) [inline eks/eks_variables.tf]: Maybe some validations here?

File: eks/eks_variables.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+variable "region" {
+  description = "Region for AWS resources"
+  type        = string
+}
+
+variable "private_route_table_id" {
+  description = "Private route table id for table associations"
+  type        = string
+}
```
Comment (erzetpe) [inline eks/eks_variables.tf:54]: Maybe some validations here?

File: eks/modules/autoscaler/variables.tf [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+variable "autoscaler_chart_version" {
+  description = "EKS chart version"
+  type        = string
+}
+
+variable "openid_connect_url" {
+  description = "OpenId connect provider url"
+  type        = string
+}
```
Comment (erzetpe) [inline eks/modules/autoscaler/variables.tf:27]: Maybe it will be good to add some validations?

File: eks/modules/control_plane/main.tf [hunk 6]
```diff
@@ -0,0 +1,59 @@
+
```
Comment (erzetpe) [inline eks/modules/control_plane/main.tf]: Is this enter needed? Have you run terraform fmt over your code from PR?

**Review by erzetpe — CHANGES_REQUESTED** [review]

File: eks/modules/nodes/variables.tf (hunk 1 above)
Comment (cicharka) [reply eks/modules/nodes/variables.tf:25]: Variables inside nested modules are reusing variables from root .tftvars so validation here is unnecessary.

File: eks/modules/autoscaler/variables.tf (hunk 5 above)
Comment (cicharka) [reply eks/modules/autoscaler/variables.tf:27]: Variables inside nested modules are reusing variables from root .tftvars so validation here is unnecessary.

File: eks/modules/control_plane/main.tf (hunk 6 above)
Comment (cicharka) [reply eks/modules/control_plane/main.tf]: Yes I have, it was never catched by `fmt`, but I will remove them.

File: eks/modules/nodes/variables.tf (hunk 2 above)
Comment (cicharka) [reply eks/modules/nodes/variables.tf:35]: Variables inside nested modules are reusing variables from root .tftvars so validation here is unnecessary.

**Comment by cicharka** [comment]
@erzetpe
Regarding validation in AKS:
- most of the values are fetched automatically from existing epiphany infrastructure
- if user wants to provide another resource group, it already should exist, therefore he cannot provide anything that was not verified by azure api
- only prefix could raise some problem as it was used in dns name, therefore I added validation only for it

Validation in EKS:
- similar situation -> we can only control prefix name
- all other resources should already exist

When it comes to the k8s specific configurations that we are allowing to tweak, users should take full responsibility of values used.

File: eks/eks_variables.tf (hunk 4 above)
Comment (cicharka) [reply eks/eks_variables.tf:54]: @erzetpe see my comment below about validation

**Review by seriva — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**